unstructured_inference/models/chipper.py

```diff
@@ -168,8 +193,81 @@ def predict(self, image) -> List[LayoutElement]:
         elements = self.format_table_elements(
             self.postprocess(image, tokens, decoder_cross_attentions),
         )
+
+        elements = self.clean_elements(elements)
+
+        return elements
+
+    def clean_elements(cls, elements):
+        """
+        Perform cleaning of empty tables and repeater elements
+        """
+        elements = cls.remove_empty_table_elements(elements)
+        elements = cls.remove_repeated_picture_elements(elements)
+        elements = cls.remove_repeated_elements(elements)
+        return elements
+
+    def remove_repeated_elements(cls, elements):
+        """
+        Removal of repeated elements. The iou and text similarity has to be large enough.
+        The first occurrence is kept since the following ones might be affected by the
+        logit processors
+        """
+        repeated_elements = []
+
+        # Get list of parents, parents should not be removed
+        parents = [e.parent for e in elements if e.parent]
+
+        for i, e1 in enumerate(elements):
+            if e1 not in repeated_elements:
+                for j in range(i + 1, len(elements)):
+                    e2 = elements[j]
+                    if e1 != e2:
+                        elements_iou = iou(e1, e2)
+                        ratio = SM(None, e1.text, e2.text).ratio()
+                        if (
+                            e1.type != e2.type
+                            and elements_iou > 0.75
+                            and ratio > 0.99
+                            and e2 not in parents
+                        ):
+                            repeated_elements.append(e2)
+
+        elements = [element for element in elements if element not in repeated_elements]
+
         return elements
 
+    @staticmethod
+    def remove_repeated_picture_elements(elements):
+        """
+        remove empty repeated picture elements
+        """
+        picture_text_to_remove = [
```

Comment:
so this list has Picture type element text that has been repeated more than 3 times. How does that translate to "empty repeated picture elements"? We could get a document with the same pictures or similar pictures with same text inside? Like a document showing examples of different fonts?